internal/cmd/controller/reconciler/bundle_controller.go

```diff
@@ -516,24 +520,28 @@ func (r *BundleReconciler) createBundleDeployment(
 }
 
 // loadBundleValues loads the values from the secret and sets them in the bundle spec
-func loadBundleValues(ctx context.Context, c client.Client, bundle *fleet.Bundle) error {
+func loadBundleValues(ctx context.Context, c client.Client, bundle *fleet.Bundle) (string, error) {
 	secret := &corev1.Secret{}
-	if err := c.Get(ctx, types.NamespacedName{Name: bundle.Name, Namespace: bundle.Namespace}, secret); err != nil {
-		return fmt.Errorf("failed to get values secret for bundle %q, this is likely temporary: %w", bundle.Name, err)
+	if err := c.Get(ctx, types.NamespacedName{Name: bundle.Name, Namespace: bundle.Namespace}, secret); apierrors.IsNotFound(err) {
```

Comment:
nit: instead of relying on 2 return values here, since non-empty message implies empty error, we could return a specific error type (e.g. a package-level `var errRetry = errors.New("retrying to get values secret for bundle")`, or even an error wrapper with a flag indicating if it's retriable), and check on L173 if the error returned by `loadBundleValues` is of that type ([errors.Is](https://pkg.go.dev/errors#Is)).